src/cbc_sdk/endpoint_standard/base.py

```diff
@@ -616,3 +697,297 @@ def _search(self, start=0, rows=0):
+
+        args = self._get_query_parameters()
+
+        url = "/api/investigate/v2/orgs/{}/enriched_events/facet_jobs".format(self._cb.credentials.org_key)
+        query_start = self._cb.post_object(url, body=args)
+        self._query_token = query_start.json().get("job_id")
+
+        self._timed_out = False
+        self._submit_time = time.time() * 1000
+
+    def _still_querying(self):
+        if not self._query_token:
+            self._submit()
+
+        status_url = "/api/investigate/v2/orgs/{}/enriched_events/facet_jobs/{}/results".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+        result = self._cb.get_object(status_url)
+        searchers_contacted = result.get("contacted", 0)
+        searchers_completed = result.get("completed", 0)
+        log.debug("contacted = {}, completed = {}".format(searchers_contacted, searchers_completed))
+        if searchers_contacted == 0:
+            return True
+        if searchers_completed < searchers_contacted:
+            if self._timeout != 0 and (time.time() * 1000) - self._submit_time > self._timeout:
+                self._timed_out = True
+                return False
+            return True
+
+        return False
+
+    def _count(self):
+        if self._count_valid:
+            return self._total_results
+
+        while self._still_querying():
+            time.sleep(.5)
+
+        if self._timed_out:
+            raise TimeoutError(message="user-specified timeout exceeded while waiting for results")
+
+        result_url = "/api/investigate/v2/orgs/{}/enriched_events/facet_jobs/{}/results".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+        result = self._cb.get_object(result_url)
+
+        self._total_results = result.get('num_found', 0)
+        self._count_valid = True
+
+        return self._total_results
+
+    def _search(self, start=0, rows=0):
+        if not self._query_token:
+            self._submit()
+
+        while self._still_querying():
+            time.sleep(.5)
+
+        if self._timed_out:
+            raise TimeoutError(message="user-specified timeout exceeded while waiting for results")
+
+        log.debug("Pulling results, timed_out={}".format(self._timed_out))
+
+        result_url = "/api/investigate/v2/orgs/{}/enriched_events/facet_jobs/{}/results".format(
+            self._cb.credentials.org_key,
+            self._query_token
+        )
+        
+        if self._limit:
+            query_parameters = {"limit": self._limit}
+        else:
+            query_parameters = {}
+
+        result = self._cb.get_object(result_url, query_parameters=query_parameters)
+        result["job_id"] = self._query_token
+        yield result
+
+    def _search_async(self):
+        """ Not very pretty hack to incorporate both synchronous and async behavior """
+        for result in self._search():
+            result.pop("job_id", None)
```

Comment:
Is it necessary to remove "job_id" before creating the EnrichedEventFacet object from the result?

Reply:
Works fine with job_id in result, so no need to remove. Fixed.